lib_eio_posix/flow.ml

```diff
@@ -0,0 +1,92 @@
+let fstat fd =
+  try
+    let ust = Low_level.fstat fd in
+    let st_kind : Eio.File.Stat.kind =
+      match ust.st_kind with
+      | Unix.S_REG  -> `Regular_file
+      | Unix.S_DIR  -> `Directory
+      | Unix.S_CHR  -> `Character_special
+      | Unix.S_BLK  -> `Block_device
+      | Unix.S_LNK  -> `Symbolic_link
+      | Unix.S_FIFO -> `Fifo
+      | Unix.S_SOCK -> `Socket
+    in
+    Eio.File.Stat.{
+      dev     = ust.st_dev   |> Int64.of_int;
+      ino     = ust.st_ino   |> Int64.of_int;
+      kind    = st_kind;
+      perm    = ust.st_perm;
+      nlink   = ust.st_nlink |> Int64.of_int;
+      uid     = ust.st_uid   |> Int64.of_int;
+      gid     = ust.st_gid   |> Int64.of_int;
+      rdev    = ust.st_rdev  |> Int64.of_int;
+      size    = ust.st_size  |> Optint.Int63.of_int64;
+      atime   = ust.st_atime;
+      mtime   = ust.st_mtime;
+      ctime   = ust.st_ctime;
+    }
+  with Unix.Unix_error (code, name, arg) -> raise @@ Err.wrap code name arg
+
+let write_bufs fd bufs =
```

Comment:
Not particular to this change but I realize we might want to consider exporting a `single_write` like `single_read`, the reason for that is that if any of the Flow write fails, the user has no way to know how many bytes have been copied before getting the exception.

Reply:
Yes; feel free to open a separate issue for that (it affects all backends).